ned/instance.py

```diff
@@ -150,42 +151,67 @@ def _ec2_config(
     return definition
 
 
-class Barrier:
+class InstanceBarrier:
     def wait(self, inst: Instance) -> Instance:
         return inst
 
 
 @dataclass(frozen=True)
-class InstanceRunningBarrier(Barrier):
+class InstanceReadyBarrier(InstanceBarrier):
```

Comment:
Would it make sense that instead of removing the one that is waiting for the instance to be in the ready state for us to have two barriers?  Since ned is a more general nitro deployment tool (not just for sequencer), are there things that we can do if the instance is running but we cannot connect to it?  (I am generally not sure.... off the top of my head, I cannot think of much)

Reply:
I thought about having two barriers. I think I thought that it would be easier to see the checks for an instance to be ready in one place in the `_is_ready` function. Since some other `ned` calls use the `inst.public_dns_name`, I thought that being able to ping the instance through that address is integral to readiness. Having said that, I think that if we see a need to split out to two barriers, it wouldn't be hard to do.